src/Features/Core/Portable/ConvertAutoPropertyToFullProperty/AbstractConvertAutoPropertyToFullPropertyCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,238 @@
+
+            var property = GetPropertyDeclaration(token);
+            if (property == null)
+            {
+                return;
+            }
+
+            if (isAbstract(property))
+            {
+                return;
+            }
+
+            // check to see if both property accessors exist and are empty
+            if (!TryGetEmptyAccessors(property, out var emptyGetAccessor, out var emptySetAccessor))
+            {
+                return;
+            }
+
+            context.RegisterRefactoring(
+                new ConvertAutoPropertyToFullPropertyCodeAction(
+                    FeaturesResources.Convert_to_full_property,
+                    c => ExpandToFullPropertyAsync(
+                        document,
+                        property,
+                        emptyGetAccessor,
+                        emptySetAccessor,
+                        root,
+                        context.CancellationToken)));
+        }
+
+        private async Task<Document> ExpandToFullPropertyAsync(
+            Document document, 
+            SyntaxNode property, 
+            SyntaxNode emptyGetAccessor, 
+            SyntaxNode emptySetAccessor, 
+            SyntaxNode root, 
+            CancellationToken cancellationToken)
+        {
+            // Get the symbol representing the property
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var propertySymbol = semanticModel.GetDeclaredSymbol(property) as IPropertySymbol;
+
+            // generate a name for the new field based on naming preferences
+            var rules = await GetNamingRulesAsync(
+                document, 
+                cancellationToken).ConfigureAwait(false);
+            var fieldName = GenerateFieldName(propertySymbol, rules);
+
+            // expand the property and add the field
+            var generator = SyntaxGenerator.GetGenerator(document);
+            var newRoot = await ExpandPropertyAndAddFieldAsync(
+                document, 
+                property, 
+                emptyGetAccessor, 
+                emptySetAccessor, 
+                root, 
+                propertySymbol, 
+                fieldName, 
+                generator, 
+                cancellationToken).ConfigureAwait(false);
+
+            return document.WithSyntaxRoot(newRoot);
+        }
+
+        private static async Task<ImmutableArray<NamingRule>> GetNamingRulesAsync(
+            Document document,
+            CancellationToken cancellationToken)
+        {
+            var options = await document.GetOptionsAsync(cancellationToken).ConfigureAwait(false);
+            var namingStyleOptions = options.GetOption(SimplificationOptions.NamingPreferences);
+            var rules = namingStyleOptions.CreateRules().NamingRules
+                .AddRange(GetDefaultRule(ImmutableArray.Create(new ModifierKind(ModifierKindEnum.IsStatic)), "s_"))
```

Comment:
We are getting all of the user-specified naming rules, then adding standard default naming rules for both static and non-static fields.  The standard naming rules are added at the end so they will only be used if the user hasn't specified a preference.  

I added this as a comment to the code.